### ISSUE TITLE
Link broken in documentation TypeScript > Type Providers

### ISSUE THREAD

**Issue by danielnmai** [fastify/website#197]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.26.2

### Node.js version

18.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

14.2.1

### Description

The `Type Providers` link leads to a not found page.
<img width="1068" alt="Screenshot 2024-03-04 at 11 02 39 AM" src="https://github.com/fastify/fastify/assets/26222469/55fc8aa1-e3db-47b5-9d2c-fd9f410a7e7d">
<img width="740" alt="Screenshot 2024-03-04 at 11 04 07 AM" src="https://github.com/fastify/fastify/assets/26222469/54034f94-b33b-4555-9d00-bbc2a0663a80">

### Steps to Reproduce

1. Go to [latest doc on TypeScript support](https://fastify.dev/docs/latest/Reference/TypeScript/#fastifyrawserver-rawrequest-rawreply-loggeropts-fastifyserveroptions-fastifyinstance)
2. Click on the `Type Providers` link
3. Observe the broken page.

### Expected Behavior

The Type Providers page should load correctly.

**Comment by matthyk** [comment]
Should be related to fastify/fastify#5308

**Comment by matthyk** [comment]
This problem appears with all documentation versions from Fastify version 4.6.x onwards

**Comment by melroy89** [comment]
Thanks for reporting I found the same issue. The links itself in markdown are correct, I mean the markdown syntax is correct and the location the link is pointing towards.

Could somebody please take a look at this? I don't know what to do.

For example, this link is broken: https://github.com/fastify/fastify/blob/main/docs/Reference/TypeScript.md?plain=1#L214

The build is showing errors:

```

 It looks like some of the broken links we found appear in many pages of your site.
Maybe those broken links appear on all pages through your site layout?
We recommend that you check your theme configuration for such links (particularly, theme navbar and footer).
Frequent broken links are linking to:
./Validation-and-Serialization.md/
./Type-Providers.md/
./HTTP2.md/
Logging.md/
./Server.md/#register
./Logging.md/

Exhaustive list of all broken links found:
```

**Comment by melroy89** [comment]
I believe something on the TypeScript page (maybe some strange character?) is breaking all the internal links on the Typescript page. Since the broken links are only present in on the TypeScript page.

**Comment by sehnsucht13** [comment]
This issue is caused by using triple backticks for inline code blocks at the start of a new line. The `Typescript.md` file in the current release contains [one instance](https://github.com/fastify/fastify/blob/v4.26.2/docs/Reference/TypeScript.md?plain=1#L65) and that breaks every relative link after it. 

docusaurus' [parser can't handle that case](https://github.com/facebook/docusaurus/issues/9876). The fix is to avoid using triple backticks for inline code blocks. Using a single backtick for inline code will work without a problem.

That specific instance in Typescript.md was removed in [PR 5361](https://github.com/fastify/fastify/pull/5361). Any future releases should have working relative links on the typescript page.  Is there any process for backporting changes to older versions of the docs? Older versions of the `4.x` release will stay broken if this is not fixed.

I found one other instance in the current docs where there triple backticks are used for inline code blocks. I'll submit a PR for that to avoid any future issues.
